### PR TITLE
Added AddItemCommandTest and DeleteItemCommandTest. Refined some exis…

### LINKED ISSUE #227: Create AddItemCommandTest class

### LINKED ISSUE #228: Create DeleteItemCommandTest class

### PR DESCRIPTION
**Pull request by ailanthustng**
…ting commands in DeleteItemCommand and SplitterModelManager to requireNonNull. Made constructor of Index class public.

### CODE REVIEW

File: src/main/java/seedu/eylah/commons/core/index/Index.java [hunk 1]
```diff
@@ -15,7 +15,7 @@
      * Index can only be created by calling {@link Index#fromZeroBased(int)} or
      * {@link Index#fromOneBased(int)}.
      */
-    private Index(int zeroBasedIndex) {
+    public Index(int zeroBasedIndex) {
```
Comment (sheexiong) [inline src/main/java/seedu/eylah/commons/core/index/Index.java:18]: Why change the private to public for the Index class?

**Review by sheexiong — COMMENTED** [review]
Here is a suggestion, maybe you could add some comment for the testing class.

File: src/main/java/seedu/eylah/commons/core/index/Index.java (hunk 1 above)
Comment (ailanthustng) [reply src/main/java/seedu/eylah/commons/core/index/Index.java:18]: I had to create a Index object for some of my test cases.

Comment (sheexiong) [reply src/main/java/seedu/eylah/commons/core/index/Index.java:18]: okay, thanks for the clarification

**Review by sheexiong — APPROVED** [review]
LGTM, thanks for the clarification for the `Index` class.

**Pull request merged**